snippets/csharp/System.Net/IWebProxy/Credentials/iwebproxy_interface.cs

```diff
@@ -66,14 +64,28 @@ public static void Main()
 
 		webProxy_Interface.Credentials = new NetworkCredential("myusername", "mypassword");
 
-		Console.WriteLine("The web proxy is : {0}", webProxy_Interface.GetProxy(new Uri("http://www.contoso.com")));
+        Uri testUri = new Uri("http://www.contoso.com");
 
 		// Determine whether the Web proxy can be bypassed for the site "http://www.contoso.com".
-		if(webProxy_Interface.IsBypassed(new Uri("http://www.contoso.com")))
+		if(webProxy_Interface.IsBypassed(testUri))
+        {
 			Console.WriteLine("Web Proxy is by passed");
+        }
 		else
-			Console.WriteLine("Web Proxy is not by passed");
+        {
+            Uri? webProxyServer = webProxy_Interface.GetProxy(testUri);
+
+            if (webProxyServer is null || webProxyServer! == testUri)
```

Comment:
If `webProxy_Interface.IsBypassed(testUri) == false`, do we still need to worry that `GetProxy` can return the URI itself, or is it always guaranteed to return `null` in that case?

Reply:
I wouldn't expect that happen, if `IsBypassed` returns false, however implementation can decide to what to do, that's why I wanted to add it.

Reply:
Yeah, if those old docs technically allowed the interface to return `destination`, maybe it's better to count for both options. But it might worth a short explanatory comment in the code sample just above this line.

Reply:
Makes sense, I'll add an explanation.